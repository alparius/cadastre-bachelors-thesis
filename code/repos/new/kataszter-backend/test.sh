#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
