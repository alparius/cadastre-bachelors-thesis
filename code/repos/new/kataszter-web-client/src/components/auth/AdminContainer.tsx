import React, { useEffect, useState } from "react";
import { FormattedMessage as FM, useIntl } from "react-intl";
import { Button, Confirm, Form, Grid, Header, Message, Segment, Select, TextArea } from "semantic-ui-react";

import UserRole from "../../data/enums/UserRole";
import IUser from "../../data/user/User";
import IUserUnderDecisionData, { NewUserUnderDecisionData } from "../../data/user/UserUnderDecisionData";
import ILabelResponse from "../../requests/response/LabelResponse";
import BoldFM from "../../util/styledMessages/BoldFM";
import CardPlaceholder from "../CardPlaceholder";
import ErrorMsg from "../ErrorMsg";

const getUserAccountSegmentColorByRole = (currentRole: UserRole) => {
    switch (currentRole) {
        case UserRole.OWNER:
            return "violet";
        case UserRole.ADMIN:
            return "blue";
        case UserRole.EDITOR:
            return "green";
        case UserRole.READER:
            return "brown";
        case UserRole.GUEST:
            return "black";
        case UserRole.REFUSED:
            return "red";
        default:
            return "orange";
    }
};

interface IAdminContainerProps {
    activeMenu: string;
    users: IUser[] | undefined;
    loading: boolean;
    fetchShowError: boolean;
    smtpDown: boolean | undefined;
    userUnderDecision: IUserUnderDecisionData;
    postResponse: ILabelResponse | undefined;
    postShowError: boolean;
    deleteResponse: ILabelResponse | undefined;
    deleteShowError: boolean;
    selfID: number | undefined;
    setUserUnderDecision(newUser: IUserUnderDecisionData): void;
    asyncPost(): Promise<void>;
    asyncDelete(url: string): Promise<void>;
}

const AdminContainer: React.FC<IAdminContainerProps> = (props: IAdminContainerProps) => {
    const {
        activeMenu,
        users,
        loading,
        fetchShowError,
        smtpDown,
        userUnderDecision,
        setUserUnderDecision,
        postResponse,
        postShowError,
        asyncPost,
        deleteResponse,
        deleteShowError,
        asyncDelete,
        selfID
    } = props;
    const { formatMessage: fm } = useIntl();

    const [confirmOpen, setConfirmOpen] = useState(false);
    const [deleteUserID, setDeleteUserID] = useState();
    const [usersToShow, setUsersToShow] = useState(users);

    useEffect(() => {
        setUsersToShow(users);
    }, [users]);

    const handleAcceptButton = (_: any, { userid }: any) => {
        setUserUnderDecision({ ...userUnderDecision, ID: userid, role: UserRole.GUEST });
    };
    const handleRefuseButton = (_: any, { userid }: any) => {
        setUserUnderDecision({ ...userUnderDecision, ID: userid, role: UserRole.REFUSED });
    };

    const handleChange = (_: any, { name, value }: any) => {
        setUserUnderDecision({ ...userUnderDecision, [name]: value });
    };

    const handleDecisionButton = (_: any) => {
        asyncPost();
        users!.filter((u: IUser) => u.ID === userUnderDecision.ID)[0].role = userUnderDecision.role;
        setUserUnderDecision(NewUserUnderDecisionData);
    };

    const handleDelete = (_: any, { userid }: any) => {
        setConfirmOpen(true);
        setDeleteUserID(userid);
    };

    const handleDeleteConfirm = () => {
        setConfirmOpen(false);
        asyncDelete(`/user/${deleteUserID}`);
        setUsersToShow(usersToShow!.filter((u: IUser) => u.ID !== deleteUserID));
        setDeleteUserID(null);
    };

    const roleOptions = [
        { key: "g", text: fm({ id: "auth.roles.guest" }), value: UserRole.GUEST },
        { key: "r", text: fm({ id: "auth.roles.reader" }), value: UserRole.READER },
        { key: "e", text: fm({ id: "auth.roles.editor" }), value: UserRole.EDITOR },
        { key: "a", text: fm({ id: "auth.roles.admin" }), value: UserRole.ADMIN }
    ];

    if (fetchShowError) {
        return <ErrorMsg errorMessage={"error.connection"} />;
    } else if (loading) {
        return <CardPlaceholder />;
    } else if (!usersToShow || usersToShow.length === 0) {
        return (
            <>
                <CardPlaceholder />
                <Header as="h2">
                    <FM id="search.noResult" />
                </Header>
            </>
        );
    } else {
        return (
            <>
                <Confirm
                    content={fm({ id: "confirm.delete.message.account" })}
                    open={confirmOpen}
                    confirmButton={fm({ id: "confirm.delete.confirm" })}
                    onConfirm={handleDeleteConfirm}
                    cancelButton={fm({ id: "confirm.delete.cancel" })}
                    onCancel={() => setConfirmOpen(false)}
                />

                {postResponse ? (
                    <Message negative={postResponse.label.startsWith("error")} positive={postResponse.label.startsWith("info")}>
                        <FM id={postResponse.label} />
                    </Message>
                ) : deleteResponse ? (
                    <Message negative={deleteResponse.label.startsWith("error")} positive={deleteResponse.label.startsWith("info")}>
                        <FM id={deleteResponse.label} />
                    </Message>
                ) : (
                    (postShowError || deleteShowError) && (
                        <Message negative>
                            <FM id="error.connection" />
                        </Message>
                    )
                )}
                {smtpDown && smtpDown === true && (
                    <Message negative>
                        <FM id="error.smtpDown" />
                    </Message>
                )}
                {usersToShow.map(
                    (user: IUser) =>
                        (activeMenu === "all" || activeMenu === user.role || (activeMenu === "admin" && user.role === UserRole.OWNER)) && (
                            <Segment key={user.ID} padded color={getUserAccountSegmentColorByRole(user.role)}>
                                <Grid stackable>
                                    <Grid.Row>
                                        <Grid.Column width="5">
                                            <FM id="auth.name_short" />: <b>{user.name}</b>
                                        </Grid.Column>
                                        <Grid.Column width="5">
                                            e-mail: <b>{user.email}</b>
                                        </Grid.Column>
                                        {user.role !== UserRole.PENDING ? (
                                            <Grid.Column width="5">
                                                <FM id="auth.role_short" />: <BoldFM id={`auth.roles.${user.role}`} />
                                            </Grid.Column>
                                        ) : (
                                            <Grid.Column width="5">
                                                <Button color="violet" inverted onClick={handleAcceptButton} userid={user.ID}>
                                                    <FM id="auth.accept" />
                                                </Button>
                                                <Button inverted secondary onClick={handleRefuseButton} userid={user.ID}>
                                                    <FM id="auth.refuse" />
                                                </Button>
                                            </Grid.Column>
                                        )}
                                        {user.ID !== selfID && user.role !== UserRole.PENDING && user.role !== UserRole.OWNER && (
                                            <Grid.Column width="1">
                                                <Button
                                                    color="red"
                                                    inverted
                                                    circular
                                                    userid={user.ID}
                                                    onClick={handleDelete}
                                                    icon="trash alternate outline"
                                                />
                                            </Grid.Column>
                                        )}
                                    </Grid.Row>
                                </Grid>
                                {userUnderDecision.ID === user.ID && (
                                    <Form>
                                        <TextArea
                                            name="message"
                                            style={{ marginTop: "1em" }}
                                            value={userUnderDecision.message}
                                            placeholder={
                                                userUnderDecision.role !== UserRole.REFUSED
                                                    ? fm({ id: "auth.message.accept" })
                                                    : fm({ id: "auth.message.refuse" })
                                            }
                                            onChange={handleChange}
                                            type="text"
                                        />
                                        {userUnderDecision.role !== UserRole.REFUSED && (
                                            <Form.Group style={{ marginTop: "1em" }} fluid>
                                                <Form.Field readOnly width="3">
                                                    <FM id="auth.role" />
                                                </Form.Field>
                                                <Form.Field
                                                    width="6"
                                                    name="role"
                                                    value={userUnderDecision.role}
                                                    placeholder={fm({ id: "auth.role" })}
                                                    onChange={handleChange}
                                                    type="select"
                                                    fluid
                                                    control={Select}
                                                    options={roleOptions}
                                                />
                                            </Form.Group>
                                        )}
                                        <Form.Button color="violet" size="large" style={{ marginTop: "1em" }} onClick={handleDecisionButton}>
                                            <FM id={userUnderDecision.role !== UserRole.REFUSED ? "auth.proceed.accept" : "auth.proceed.refuse"} />
                                        </Form.Button>
                                    </Form>
                                )}
                            </Segment>
                        )
                )}
                {userUnderDecision.ID !== 0 && <div style={{ height: "5em" }} />}
            </>
        );
    }
};

export default AdminContainer;
