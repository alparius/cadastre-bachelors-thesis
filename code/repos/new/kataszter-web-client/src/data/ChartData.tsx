interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
