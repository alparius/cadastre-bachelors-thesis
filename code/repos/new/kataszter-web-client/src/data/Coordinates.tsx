interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
