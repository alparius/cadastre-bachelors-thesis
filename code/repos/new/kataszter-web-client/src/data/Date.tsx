interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
