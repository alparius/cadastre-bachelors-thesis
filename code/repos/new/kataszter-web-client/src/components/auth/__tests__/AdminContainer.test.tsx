/*===================
  AdminContainer.unit.test.tsx
 ===================*/
import React from "react";

import IUser from "../../../data/user/User";
import { NewTestLabelResponseError, NewTestLabelResponseInfo, NewTestUser, NewTestUserUnderDecisionData } from "../../../util/test/data/test-data";
import { createComponentWithIntl, createComponentWithIntlAndRouter } from "../../../util/test/test-wrapper";
import AdminContainer from "../AdminContainer";

const users: IUser[] = [NewTestUser];

const handlers = {
    userUnderDecision: NewTestUserUnderDecisionData,
    activeMenu: "",
    selfID: 678,
    setUserUnderDecision: jest.fn(),
    asyncPost: jest.fn(),
    asyncDelete: jest.fn()
};

describe("AdminContainer element", () => {
    it("should render a segment group when the users array is not empty", () => {
        const container = createComponentWithIntl(
            <AdminContainer
                users={users}
                loading={false}
                fetchShowError={false}
                postShowError={false}
                deleteShowError={false}
                deleteResponse={undefined}
                postResponse={undefined}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder card and a message when the users array is empty", () => {
        const emptyUsers: IUser[] = [];
        const container = createComponentWithIntl(
            <AdminContainer
                users={emptyUsers}
                loading={false}
                fetchShowError={false}
                postShowError={false}
                deleteShowError={false}
                deleteResponse={undefined}
                postResponse={undefined}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when fetchShowError is true", () => {
        const container = createComponentWithIntlAndRouter(
            <AdminContainer
                users={users}
                loading={false}
                fetchShowError
                postShowError={false}
                deleteShowError={false}
                deleteResponse={undefined}
                postResponse={undefined}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should also render an error message when postShowError is true", () => {
        const container = createComponentWithIntl(
            <AdminContainer
                users={users}
                loading={false}
                fetchShowError={false}
                postShowError
                deleteShowError={false}
                deleteResponse={undefined}
                postResponse={undefined}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should also render an error message when deleteShowError is true", () => {
        const container = createComponentWithIntl(
            <AdminContainer
                users={users}
                loading={false}
                fetchShowError={false}
                postShowError={false}
                deleteShowError
                deleteResponse={undefined}
                postResponse={undefined}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError and loading are both true", () => {
        const container = createComponentWithIntlAndRouter(
            <AdminContainer
                users={users}
                loading
                fetchShowError
                postShowError
                deleteShowError
                deleteResponse={undefined}
                postResponse={undefined}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder card when only loading is true", () => {
        const container = createComponentWithIntl(
            <AdminContainer
                users={users}
                loading
                fetchShowError={false}
                postShowError={false}
                deleteShowError={false}
                deleteResponse={undefined}
                postResponse={undefined}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should also render a message when response exists", () => {
        const container = createComponentWithIntl(
            <AdminContainer
                users={users}
                loading
                fetchShowError={false}
                postShowError={false}
                deleteShowError={false}
                deleteResponse={NewTestLabelResponseInfo}
                postResponse={undefined}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should also render an error when a bad response exists", () => {
        const container = createComponentWithIntl(
            <AdminContainer
                users={users}
                loading
                fetchShowError={false}
                postShowError={false}
                deleteShowError={false}
                deleteResponse={undefined}
                postResponse={NewTestLabelResponseError}
                {...handlers}
            />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });
});
