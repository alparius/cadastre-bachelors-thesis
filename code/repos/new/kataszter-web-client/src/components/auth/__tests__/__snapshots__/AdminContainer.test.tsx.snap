// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminContainer element should also render a message when response exists 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="header"
        >
          <div
            className="line"
          />
        </div>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AdminContainer element should also render an error message when deleteShowError is true 1`] = `
<div
  className="ui negative message"
>
  Could not estabilish server connection.
</div>
`;

exports[`AdminContainer element should also render an error message when postShowError is true 1`] = `
<div
  className="ui negative message"
>
  Could not estabilish server connection.
</div>
`;

exports[`AdminContainer element should also render an error when a bad response exists 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="header"
        >
          <div
            className="line"
          />
        </div>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AdminContainer element should render a placeholder card and a message when the users array is empty 1`] = `
Array [
  <div
    className="ui fluid card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="header"
          >
            <div
              className="line"
            />
          </div>
        </div>
      </div>
      <div
        className="description"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
        </div>
      </div>
    </div>
  </div>,
  <h2
    className="ui header"
  >
    No results were found for your search. Try widening the search criterias.
  </h2>,
]
`;

exports[`AdminContainer element should render a placeholder card when only loading is true 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="header"
        >
          <div
            className="line"
          />
        </div>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AdminContainer element should render a segment group when the users array is not empty 1`] = `null`;

exports[`AdminContainer element should render an error message when fetchShowError is true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;

exports[`AdminContainer element should render an error message when showError and loading are both true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;
